server/http.go

```diff
@@ -127,18 +128,35 @@ func (h *httpCache) handleGetValidAC(w http.ResponseWriter, r *http.Request, has
 		return
 	}
 
-	w.Header().Set("Content-Type", "application/octet-stream")
-	w.Header().Set("Content-Length", strconv.FormatInt(int64(len(data)), 10))
-	bytesWritten, err := w.Write(data)
+	if r.Header.Get("Accept") == "application/json" {
+		ar := &pb.ActionResult{}
+		err = proto.Unmarshal(data, ar)
+		if err != nil {
+			h.logResponse(http.StatusInternalServerError, r)
+			return
+		}
 
-	if err != nil {
-		h.logResponse(http.StatusInternalServerError, r)
-		return
-	}
+		w.Header().Set("Content-Type", "application/json")
+		marshaler := jsonpb.Marshaler{}
+		err = marshaler.Marshal(w, ar)
+		if err != nil {
+			h.logResponse(http.StatusInternalServerError, r)
+			return
+		}
```

Comment:
Maybe just return at the end of this block, then we don't need the "else" (handle the protobuf case at one indent level less).